src/Integration.UnitTests/MefServices/SharedBindingSuggestionServiceTests.cs

```diff
@@ -127,6 +130,20 @@ public void Dispose_UnsubscribesFromActiveSolutionChanged()
         activeSolutionTracker.Received(1).ActiveSolutionChanged -= Arg.Any<EventHandler<ActiveSolutionChangedEventArgs>>();
     }
 
+    [TestMethod]
+    public void ActiveSolutionChanged_SolutionIsOpened_ShowsGoldBarAndShowManageBindingDialog()
+    {
+        MockSharedBindingConfigExists();
+        MockSolutionMode(SonarLintMode.Standalone);
+        suggestSharedBindingGoldBar.When(x => x.Show(ServerType.SonarQube, Arg.Any<Action>())).Do(callInfo =>
+        {
+            callInfo.Arg<Action>()();
+            connectedModeManager.Received(1).ShowManageBindingDialog(true);
```

Comment:
Not sure if it's a good idea to put assertions inside setup code. If you wanted to assert the order of calls, you can use Received.InOrder(...)

Reply:
I don't want to test the order, but I want to know what happens when the action is called. So, I really have to store the action and call it. I can refactor this to make it more explicit.